src/ForeignController.sol

```diff
@@ -214,6 +216,67 @@ contract ForeignController is AccessControl {
         }
     }
 
+    /**********************************************************************************************/
+    /*** Relayer ERC4626 functions                                                              ***/
+    /**********************************************************************************************/
+
+    function depositERC4626(address vault, uint256 amount)
+        external
+        onlyRole(RELAYER)
+        isActive
+        rateLimited(
+            RateLimitHelpers.makeAssetKey(LIMIT_VAULT_DEPOSIT, vault),
```

Comment:
We should use another term instead of vault across the code here to avoid conflating with the allocation system, I used `token` in the mainnet implementation, not sure what you think of that